var/spack/repos/builtin/packages/gcc/package.py

```diff
@@ -71,10 +71,17 @@ class Gcc(AutotoolsPackage):
 
     # https://gcc.gnu.org/install/prerequisites.html
     depends_on('gmp@4.3.2:')
-    depends_on('mpfr@2.4.2:')
+    # GCC 7.3 does not compile with newer releases on some platforms, see
+    #   https://github.com/spack/spack/issues/6902#issuecomment-433030376
+    depends_on('mpfr@2.4.2:3.1.6')
     depends_on('mpc@0.8.1:', when='@4.5:')
-    depends_on('isl@0.14', when='@5:5.9')
-    depends_on('isl@0.15:', when='@6:')
+    # Already released GCC versions do not support any newer version of ISL
+    #   GCC 5.4 https://github.com/spack/spack/issues/6902#issuecomment-433072097
+    #   GCC 7.3 https://github.com/spack/spack/issues/6902#issuecomment-433030376
+    #   GCC 9+  https://gcc.gnu.org/bugzilla/show_bug.cgi?id=86724
+    depends_on('isl@0.15', when='@5:5.9')
+    depends_on('isl@0.15:0.18', when='@6:8.9')
+    depends_on('isl@0.15:0.20', when='@9:')
```

Comment:
>> when=@9: constraint appears to be speculative

> agreed, thx!

What I meant here was that it may be overly-constraining to set this here. It may be confusing to users where it comes from, since https://gcc.gnu.org/bugzilla/show_bug.cgi?id=86724 doesn't mention gcc 9. My interpretation is that you infer this will be appropriate based on that report. I think the comment should add clarification:

`GCC up to 8.2.0 does not work with isl@20, this was fixed as part of the gcc 8 development branch and so gcc@9: is anticipated to work with isl up to version 20 (and perhaps higher)`

Reply:
The link mentions GCC 9 in `Known to work` on the top right hand side :)